pandarallel/core.py

```diff
@@ -321,10 +321,16 @@ def closure(
                     progress_bars.set_error(worker_index)
                     progress_bars.update(progresses)
 
-            return wrapped_reduce_function(
-                (Path(output_file.name) for output_file in output_files),
-                reduce_extra,
-            )
+            try:
+                return wrapped_reduce_function(
+                    (Path(output_file.name) for output_file in output_files),
+                    reduce_extra,
+                )
+            except EOFError:
+                # Loading the files failed, this most likely means that there
+                # was some error during processing and the files were never
+                # saved at all. 
+                results_promise.get()
 
```

Comment:
There should maybe be another `raise` here to raise to original Error in case nothing actually went wrong with the `starmap_async` call.